test/unit/deferred.js

```diff
@@ -525,6 +525,34 @@ QUnit.test( "jQuery.Deferred.then - spec compatibility", function( assert ) {
 	} catch ( _ ) {}
 } );
 
+QUnit[ window.console ? "test" : "skip" ]( "jQuery.Deferred.exceptionHook", function( assert ) {
+
+	assert.expect( 1 );
+
+	var done = assert.async(),
+		defer = jQuery.Deferred(),
+		oldWarn = window.console.warn;
+
+	window.console.warn = function( msg ) {
```

Comment:
> sinon spy but they didn't offer that much value for such a simple case

Agreed. A few years ago I tried it and saw a bloat library polluting the global scope and QUnit's api. At that moment I've made [DexterJS](https://github.com/leobalter/dexterjs) with a very simple spy/stub API alternative but it's not maintained since I didn't have many contributions and stopped using mocks.